src/main/resources/templates/main.html

```diff
@@ -1,47 +1,96 @@
 <!DOCTYPE html>
 <html xmlns:th="http://www.thymeleaf.org">
 <head>
-<meta charset="UTF-8">
-<title th:text="#{main.title}"></title>
+	<meta charset="UTF-8">
+	<title th:text="#{main.title}"></title>
+	<style>
+		.embedCode {
+			font-family: 'Courier New';
+			border: 1px solid;
+			border-color: black;
+			padding: 5px;
+			width: 500px;
+			height: 80px;
+			font-size:10pt
+		}
+		#chartIFrame {
+			width:95%;
+			height:1000px;
+			margin: 20px;
+			border:0;
+		}
+	</style>
+	<script>
+		function showHideEmbedCode(embedChartId, embedChartButtonId) {
+			var embedChartStyle = document.getElementById(embedChartId).style;
+			embedChartStyle.display = (embedChartStyle.display === "none" ? "block" : "none");
+			document.getElementById(embedChartButtonId).textContent = (embedChart.style.display === "none" ? "Embed chart" : "Hide embed code");
+		}
+		function changeChart(chartFrameId, chartSelectorId) {
+			document.getElementById('chartIFrame').setAttribute('src', document.getElementById('chartSelector').value);
+			document.getElementById('embedChart').textContent =
+				'<iframe width="600" height="400" seamless frameBorder="0" scrolling="no" src="' + document.getElementById('chartSelector').value + '&height=400"></iframe>';
+		}
+	</script>
 </head>
 <body>
 	<h1 th:text="#{main.title}"></h1>
 	<p>
 		<a th:href="${data.fileUrl}" th:text="${data.fileName}"></a>
 		<span th:text="${'('+ data.fileSize + ')'}"></span>
 	</p>
-	<table th:if="${!data.columns.isEmpty()}">
-		<thead>
-			<tr>
-				<th>Column</th>
-				<th>Type</th>
-			</tr>
-		</thead>
-		<tbody>
-			<tr th:each="column: ${data.columns}">
-				<td th:text="${column.name}"></td>
-				<td th:text="${column.type}"></td>
-			</tr>
-		</tbody>
-	</table>
 
 	<th:block th:switch="${data.status.name()}">
 		<div th:case="'READY'">
+			<table th:if="${!data.columns.isEmpty()}">
+				<thead>
+				<tr>
+					<th>Column</th>
+					<th>Type</th>
+				</tr>
+				</thead>
+				<tbody>
+				<tr th:each="column: ${data.columns}">
+					<td th:text="${column.name}"></td>
+					<td th:text="${column.type}"></td>
+				</tr>
+				</tbody>
+			</table>
 			<form method="POST" action="#" th:action="@{/dataverse-superset}">
 				<input type="hidden" id="name" name="datasetName" th:value="${data.getName()}">
 				<button type="submit">Import to Superset</button>
 			</form>
 		</div>
+
 		<div th:case="'IN_PROGRESS'">
 			<p>Import to Superset in progress... <button onClick="window.location.reload();">Refresh</button></p>
 		</div>
+
 		<div th:case="'COMPLETE'">
 			<p>
-				Import to Superset complete! ID: <a
+				Dataset imported to superset! Explore the data:  <a
```

Comment:
Let's move the superset link under the charts, which are more important. Also, it could be a button labeled "Create a chart in Superset", opening in a new card.

Reply:
The same link as "Explore the data", so I just moved in below.